src/alchemlyb/estimators/ti_.py

```diff
@@ -166,3 +177,129 @@ def separate_dhdl(self):
                 new.attrs = self.dhdl.attrs
                 dhdl_list.append(new)
         return dhdl_list
+
+    def gaussian_quadrature(self, dHdl, means, variances):
+        """
+        numerically estimate the free energy using gaussian quadrature with 
+        lambda values suggested in Amber manual
+
+        Parameters
+        ----------
+        dHdl : DataFrame
+        means : DataFrame
+        variances: DataFrame
+        """
+        # extract the lambda values used in the simulations
+        lambdas = means.reset_index()[means.index.names[:]].iloc[:].values.sum(axis=1)
+        num_lambdas = len(lambdas)
+        # check if the lambda values used match the common ones suggested in Amber manual and assign cooresponding weights
+        if num_lambdas==1:
```

Comment:
Avoid code repetition and that many elif statements. Make the code much more compact and extendable by storing the special points and weights in a data structure
```
special_points = {1:   [[0.5], [1.0]], 
                             2:  [[0.21132, 0.78867], [0.5, 0.5]],
                             3:  [[0.1127, 0.5, 0.88729],  [0.27777, 0.44444, 0.27777]]
                             ...}
``` 
and have one code path that pulls its points from the `special_points`.